etcdctlv3/README.md

```diff
@@ -109,6 +109,61 @@ OK
 ./etcdctl range foo
 ```
 
+### TXN [options]
+
+Watch watches events stream on keys or prefixes. The watch command runs until it encounters an error or is terminated by the user.
+
+#### Options
+
+- hex -- print out keys and values as hex encoded string
+
+- i -- input transaction with interactive mode
+
+#### Input Format
+```bnf
+<Txn> ::= <CMP>* "\n" <THEN> "\n" <ELSE> "\n"
+<CMP> ::= <key> <CMPOP> <(CMPCREATE|CMPMOD|CMPVAL|CMPVER)>
+<CMPOP> ::= "g" | "greater" | "e" | "equal" | "l" | "less"
+<CMPCREATE> :=("c"|"create") <revision>
+<CMPMOD> ::= ("m"|"mod") <revision>
+<CMPVAL> ::= ("val"|"value") <value>
+<CMPVER> ::= ("ver"|"version") <version>
+<THEN> ::= Op*
+<ELSE> ::= Op*
+<OP> ::= see put, get, del etcdctl command syntax
+```
+
+#### Return value
+
+Simple reply
+
+- SUCCESS if etcd processed the transaction success list, FAILURE if etcd processed the transaction failure list.
+
+- Simple reply for each command executed request list.
+
+- Additional error string if TXN failed. Exit code is non-zero.
+
+TODO: probably json and binary encoded proto
+
+#### Examples
+
+``` bash
+./etcdctl txn -i
+key1 greater mod 0
+
+put key1 "overwrote-key1"
```

Comment:
for the interactive mode, probably we can print out hints? Like cmp:\n, success: \n, failure: \n?